spring-cloud-starter-stream-sink-sftp/src/main/java/org/springframework/cloud/stream/app/sftp/sink/SftpSinkConfiguration.java

```diff
@@ -58,4 +58,25 @@ public IntegrationFlow ftpInboundFlow(SftpSinkProperties properties, SessionFact
 			.get();
 	}
 
+	@Bean
+	@ConditionalOnMissingBean
+	public SessionFactory<LsEntry> sftpSessionFactory(SftpSinkProperties properties) {
+		DefaultSftpSessionFactory sftpSessionFactory = new DefaultSftpSessionFactory();
+		sftpSessionFactory.setHost(properties.getFactory().getHost());
+		sftpSessionFactory.setPort(properties.getFactory().getPort());
+		sftpSessionFactory.setUser(properties.getFactory().getUsername());
+		sftpSessionFactory.setPassword(properties.getFactory().getPassword());
+		sftpSessionFactory.setAllowUnknownKeys(properties.getFactory().isAllowUnknownKeys());
+		if (properties.getFactory().getKnownHostsExpression() != null) {
+			sftpSessionFactory.setKnownHosts("#{" + properties.getFactory().getKnownHostsExpression() + "}");
```

Comment:
I didn't look at the code, I just moved it - but you're right - no idea which brain cell generated that, but it must have long ago - it looks like a bad port from XD.